### ISSUE TITLE
chapters for all youtube videos

### ISSUE THREAD

**Issue by adamjonas** [bitcointranscripts/tstbtc#74]
Youtube videos that don't have key moments explicitly left in the description still have chapters that are auto-generated by the platform. We should be taking advantage of those chapters to create markdown headers in our transcripts.

For example, the video https://www.youtube.com/watch?v=NT_dMqB1xuA has the following chapters in the metadata and the `-C` should create them accordingly.

```
...
"chapters": [{"start_time": 0.0, "end_time": 347.0, "title": "Path finding on the Lightning Network (gossip)"}, {"start_time": 347.0, "end_time": 744.0, "title": "Comparing IP Routing with Onion Routing"}, {"start_time": 744.0, "end_time": 818.0, "title": "Key ingredient for routing is to know a path"}, {"start_time": 818.0, "end_time": 886.0, "title": "Realty Check! Weights are dynamic"}, {"start_time": 886.0, "end_time": 950.0, "title": "Realty Check! Topology changes during pathfinding"}, {"start_time": 950.0, "end_time": 972.0, "title": "Realty Check! Dynamic problem with uncertainty"}, {"start_time": 972.0, "end_time": 1022.0, "title": "Purpose of the Gossip Protocol"}, {"start_time": 1022.0, "end_time": 1287.0, "title": "4 announcement messages"}, {"start_time": 1287.0, "end_time": 1414.0, "title": "Querying the gossip protocol for information"}, {"start_time": 1414.0, "end_time": 1534.0, "title": "Summary of topology information for pathfinding"}, {"start_time": 1534.0, "end_time": 1640.0, "title": "Strategies for path finding"}, {"start_time": 1640.0, "end_time": 1961.0, "title": "Considerations when probing for paths"}, {"start_time": 1961.0, "end_time": 2574.0, "title": "Considerations with AMP"}, {"start_time": 2574.0, "end_time": 2961.0, "title": "Short Intro to Justin Time Routing"}, {"start_time": 2961.0, "end_time": 3031.0, "title": "Considerations with JIT"}, {"start_time": 3031.0, "end_time": 3153.0, "title": "(Multihop) Trampoline payments"}, {"start_time": 3153.0, "end_time": 3288.0, "title": "General thoughts on pathfinding on Lightning"}, {"start_time": 3288.0, "end_time": 3306, "title": "Path finding has many open challenges!"}]
...
```

**Comment by masterchief164** [comment]
As far as I can see we can't directly extract the auto-generated chapters from youtube. We'll have to switch from pytube to youtube-dlp. This means we'll need a complete change in the dependencies and many changes in the code base.

**Comment by adamjonas** [comment]
Can we not just make a YouTube-DLP call just for metadata when we are passed the flag? Seems pretty inexpensive.

**Comment by masterchief164** [comment]
I guess this issue has been resolved as well by #89
